plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php

```diff
@@ -32,81 +29,82 @@ public function watchFolder(KalturaDropFolder $dropFolder)
 
 		$vendorIntegrationSetting = $this->vendorPlugin->vendorIntegration->get($dropFolder->integrationId);
 		KalturaLog::info('Watching folder ['.$this->dropFolder->id.']');
-		$driveLocationsDir =  KBatchBase::$taskConfig->params->teams->driveLocationsDir;
-		$driveLocationsFileNamePrefix = KBatchBase::$taskConfig->params->teams->driveLocationsFileNamePrefix;
 
-		$existingDropFolderFiles = $this->loadDropFolderFiles();
+        $existingDropFolderFiles = $this->loadDropFolderFiles();
 
-		$files = scandir($driveLocationsDir);
+		// Retrieve all user custom metadata objects for the metadata profile specified on the drop folder's integration ID
+        $userMetadataSearchItem = new KalturaMetadataSearchItem();
+        $userMetadataSearchItem->metadataProfileId = $vendorIntegrationSetting->userMetadataProfileId;
+        $userMetadataSearchItem->orderBy = self::ASC_LAST_ACCESSED_DATE;
 
-		$currentFile = null;
-		foreach ($files as $file)
-		{
-			if ($file == '.' || $file == '..' || strpos($file,self::LOCKED_FILE_MIDFIX) || strpos($file, $driveLocationsFileNamePrefix . $this->dropFolder->id) === false)
-			{
-				continue;
-			}
-			else
-			{
-				$currentFile = $file;
-				break;
-			}
-		}
-
-		if (!$currentFile)
-		{
-			KalturaLog::info('No iterable file found. Skipping.');
-			return;
-		}
-
-		KalturaLog::info("On this iteration, processing file $currentFile");
+        $condition = new KalturaSearchMatchCondition();
+        $condition->field = self::OPT_IN_XPATH;
+        $condition->value = self::OPT_IN_VALUE;
 
-		$currentFilePieces = explode('.', $currentFile, 2);
-		$lockedFileName = $currentFilePieces[0] . self::LOCKED_FILE_MIDFIX . $currentFilePieces[1];
+        $userMetadataSearchItem->items = [ $condition ];
 
-		rename($driveLocationsDir . DIRECTORY_SEPARATOR . $currentFile, $driveLocationsDir . DIRECTORY_SEPARATOR . $lockedFileName);
+        $userFilter = new KalturaUserFilter();
+        $userFilter->advancedSearch = $userMetadataSearchItem;
 
-		$itemCount = 0;
+        $pager = new KalturaFilterPager();
+        $pager->pageIndex = 1;
+        $pager->pageSize = KBatchBase::$taskConfig->params->teams->userPageSize;
 
-		$drivesFileHandle = fopen($driveLocationsDir . DIRECTORY_SEPARATOR . $lockedFileName, 'r+');
+        KBatchBase::$kClient->setResponseProfile($this->constructResponseProfile($vendorIntegrationSetting->userMetadataProfileId));
 
-		$newDriveUrlAssoc = array();
-		$line = fgetcsv($drivesFileHandle);
+        $response = KBatchBase::$kClient->user->listAction($userFilter, $pager);
 
-		while ($line)
+		foreach ($response->objects as $user)
 		{
-			list ($userId, $driveLastPageUrl) = $line;
-			KalturaLog::info("Handling drive for user ID: $userId, drive URL: $driveLastPageUrl");
-			$items = $this->graphClient->sendGraphRequest($driveLastPageUrl);
+		    /* @var $user KalturaUser  */
+            $userMetadataObj = $user->relatedObjects[self::MS_GRAPH_METADATA]->objects[0];
```

Comment:
is it always initialized? $user->relatedObjects[self::MS_GRAPH_METADATA]

Reply:
@inbal-ben-david it comes from the response profile which is initialized and attached to the client on line 56.